numpy/testing/_private/extbuild.py

```diff
@@ -167,7 +168,14 @@ def _make_methods(functions, modname):
 def _make_source(name, init, body):
     """ Combines the code fragments into source code ready to be compiled
     """
-    code = """
+    # python.h doesn't set Py_GIL_DISABLED on the windows free-threade build
```

Comment:
The reason this is needed here is because we're using `shared_module` in the generated `meson.build` code here. And I think the reason we're doing that, rather than the expected `py.extension_module`, is ?? (can't remember this at all). Meson would take care of inserting this define if we were building an extension module.

Reply:
I made the change from setup.py to meson in #24153 and didn't give too much thought to `shared_module` vs. `py.extension_module`.

Reply:
Ah okay, thanks for the context @mattip. @ngoldbaum I'd suggest to switch to `py.extension_module` then, since that's much more realistic and then shouldn't need this workaround (except maybe with GCC right now, because Meson 1.6.1 isn't yet released and will contain one fix specific to MinGW).